rosbag2_cpp/src/rosbag2_cpp/clocks/time_controller_clock.cpp

```diff
@@ -168,7 +168,9 @@ bool TimeControllerClock::sleep_until(rcutils_time_point_value_t until)
       impl_->cv.wait_for(lock, impl_->sleep_time_while_paused);
     } else {
       const auto steady_until = impl_->ros_to_steady(until);
-      impl_->cv.wait_until(lock, steady_until);
+      if (steady_until > impl_->now_fn()) {
```

Comment:
i do not think we need `else` statement here, instead it should fall down to the current sequence.

Reply:
@fujitatomoya Sorry I didn't get. What `else` statement did you mean?
Also it would be nice to add some comment above why we do this trick. To avoid removal it by someone in the future.

Reply:
ah, that was supposed to mean, in `else` statement it does not return `true`, instead it honors `pause` state. (that was originally a note for myself, but thanks for checking.)